### PR TITLE
Uniformize pagination usage

### PR DESCRIPTION
**Pull request by BenFradet**

### CODE REVIEW

File: github4s/src/main/scala/github4s/algebras/Issues.scala [hunk 1]
```diff
@@ -275,6 +283,7 @@ trait Issues[F[_]] {
       repo: String,
       number: Int,
       label: String,
+      pagination: Option[Pagination] = None,
```
Comment (dcsobral) [inline github4s/src/main/scala/github4s/algebras/Issues.scala:286]: Field was not added to scaladoc.

File: github4s/src/main/scala/github4s/interpreters/GitDataInterpreter.scala [hunk 2]
```diff
@@ -30,6 +30,7 @@ class GitDataInterpreter[F[_]](implicit client: HttpClient[F], accessToken: Opti
       owner: String,
       repo: String,
       ref: String,
+      pagination: Option[Pagination] = None,
```
Comment (dcsobral) [inline github4s/src/main/scala/github4s/interpreters/GitDataInterpreter.scala:33]: It needs to be used on the `get` call.

File: github4s/src/main/scala/github4s/interpreters/IssuesInterpreter.scala [hunk 3]
```diff
@@ -32,6 +32,7 @@ class IssuesInterpreter[F[_]](implicit client: HttpClient[F], accessToken: Optio
   override def listIssues(
       owner: String,
       repo: String,
+      pagination: Option[Pagination] = None,
```
Comment (dcsobral) [inline github4s/src/main/scala/github4s/interpreters/IssuesInterpreter.scala:35]: It needs to be used on the `get` call.

File: github4s/src/main/scala/github4s/interpreters/IssuesInterpreter.scala [hunk 4]
```diff
@@ -96,6 +97,7 @@ class IssuesInterpreter[F[_]](implicit client: HttpClient[F], accessToken: Optio
       owner: String,
       repo: String,
       number: Int,
+      pagination: Option[Pagination] = None,
```
Comment (dcsobral) [inline github4s/src/main/scala/github4s/interpreters/IssuesInterpreter.scala:100]: It needs to be used on the `get` call.

File: github4s/src/main/scala/github4s/interpreters/IssuesInterpreter.scala [hunk 5]
```diff
@@ -155,6 +157,7 @@ class IssuesInterpreter[F[_]](implicit client: HttpClient[F], accessToken: Optio
...
```
Comment (dcsobral) [inline github4s/src/main/scala/github4s/interpreters/IssuesInterpreter.scala:160]: The `pagination` parameter was added but not used anywhere. That is a problem in itself, but it also indicates another issue: it isn't being tested.

**Comment by BenFradet** [comment]
Thanks for the review 👍 , went mindlessly through the algebras

**Comment by BenFradet** [comment]
I think this is good to go now :+1:

**Comment by dcsobral** [comment]
@BenFradet This will conflict with #462 , and introduces more cases of #449 . It will be easier if this is cleared of that issue, merged, and then #462 rebased on top of it.

**Comment by BenFradet** [comment]
#462 is against the branch associated to this PR for that reason :+1:

**Pull request merged**